### PR TITLE
refacto(Connections): hide sensitive informations when printing  connection

### PR DESCRIPTION
**Pull request by cheikhgwane**
Relative to [issue](https://github.com/BLSQ/openhexa-team/issues/512) : hide sensitive informations when printing  connection

### CODE REVIEW

File: openhexa/sdk/workspaces/connection.py [hunk 1]
```diff
@@ -7,6 +7,9 @@ class DHIS2Connection:
     username: str
     password: str
 
+    def __repr__(self):
+        return f"DHIS2Connection(url='{self.url}', username='{self.username}', password='*********')"
```
Comment (qgerome) [inline openhexa/sdk/workspaces/connection.py:11]: Just don't add the password to the repr, it's easier and do not try to convey a message (that can be false)

File: openhexa/sdk/workspaces/connection.py [hunk 2]
```diff
@@ -16,6 +19,9 @@ class PostgreSQLConnection:
     password: str
     database_name: str
 
+    def __repr__(self):
+        return f"PostgreSQLConnection(host='{self.host}', port='{self.port}', username='{self.username}', password='*********', database_name='{self.database_name}')"
```
Comment (qgerome) [inline openhexa/sdk/workspaces/connection.py:23]: Remove the password, it's easier.

**Review by qgerome — CHANGES_REQUESTED** [review]

File: openhexa/sdk/workspaces/connection.py (hunk 1 above)
Comment (cheikhgwane) [reply openhexa/sdk/workspaces/connection.py:11]: So 
```python
def __repr__(self):
   return f"DHIS2Connection(url='{self.url}', username='{self.username}')"
```
?

File: tests/test_workspace.py [hunk 3]
```diff
@@ -116,6 +124,8 @@ def test_workspace_gcs_connection():
         s3_connection = workspace.gcs_connection(identifier=identifier)
         assert s3_connection.service_account_key == service_account_key
         assert s3_connection.bucket_name == bucket_name
+        assert re.search("service_account_key", repr(s3_connection)) is None
+        assert re.search("service_account_key", str(s3_connection)) is None
 
 
 def test_workspace_custom_connection():
```
Comment (pvanliefland) [inline tests/test_workspace.py:131]: Would it be possible to have a test for secret values in custom connection as well?

**Review by pvanliefland — CHANGES_REQUESTED** [review]
Hi @cheikhgwane , almost there...

File: tests/test_workspace.py (hunk 3 above)
Comment (cheikhgwane) [reply tests/test_workspace.py:131]: Okay

**Review by pvanliefland — APPROVED** [review]
LGTM!

**Pull request merged**